src/3d/qgscolorramptexture.h

```diff
@@ -0,0 +1,73 @@
+/***************************************************************************
+                         qgscolorramptexture.h
+                         -------------------------
+    begin                : january 2020
+    copyright            : (C) 2020 by Vincent Cloarec
+    email                : vcloarec at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSCOLORRAMPTEXTURE_H
+#define QGSCOLORRAMPTEXTURE_H
+
+/// @cond PRIVATE
+
+#include <Qt3DRender/QEffect>
+#include <Qt3DRender/QGraphicsApiFilter>
+#include <Qt3DRender/QParameter>
+#include <Qt3DRender/QTexture>
+
+#include <QUrl>
+#include <QVector2D>
+#include <QVector3D>
+#include <QVector4D>
+#include <Qt3DRender/QBuffer>
+#include <QByteArray>
+
+#include "qgsmeshlayer.h"
+#include "qgsmeshlayerutils.h"
+#include "qgstriangularmesh.h"
+
+class ColorRampTextureGenerator: public Qt3DRender::QTextureImageDataGenerator
```

Comment:
as this classes are now in header file, maybe it is better to change the name with `QgsColorRampTextureGenerator`?

Reply:
Ah, sorry I forgot to change that.

Reply:
BTW, do you think I need to use verticalScale parameter in QgsColorRampTexture ?

Reply:
is vertical scale is a setting available for point cloud?

Reply:
Not really it but I think the height thing is equivalent